src/server/cluster/incoming_slot_migration.cc

```diff
@@ -21,46 +22,62 @@ using absl::GetFlag;
 // It is created per shard on the target node to initiate FLOW step.
 class ClusterShardMigration {
  public:
-  ClusterShardMigration(uint32_t shard_id, Service* service) : source_shard_id_(shard_id) {
-    executor_ = std::make_unique<JournalExecutor>(service);
+  ClusterShardMigration(uint32_t shard_id, Service* service)
+      : source_shard_id_(shard_id), socket_(nullptr), executor_(service) {
```

Comment:
nit: it's safer to init raw values like `socket_` in their declaration rather than here

Reply:
In this case I can not write ABSL_GUARDED_BY(mu_);

Reply:
It's not intuitive, but you can. I think it goes like this:
```
type name_ ABSL_GUARDED_BY(mu_) = default_value;
```

Reply:
I don't like this) may I skip this?

Reply:
I personally think it's worth the added safety, but I never insist on my reviews as long as both sides agree on the facts :)
Let me have one last pass and I'll approve